corehq/apps/reports_core/filters.py

```diff
@@ -277,16 +277,19 @@ class ChoiceListFilter(BaseFilter):
     """
     template = 'reports_core/filters/choice_list_filter.html'
 
-    def __init__(self, name, datatype, label='Choice List Filter',
+    def __init__(self, name, field, datatype, label='Choice List Filter',
                  css_id=None, choices=None):
+        from corehq.apps.userreports.reports.filters.choice_providers import StaticChoiceProvider
         params = [
             FilterParam(name, True),
         ]
         super(ChoiceListFilter, self).__init__(name=name, params=params)
+        self.field = field
         self.datatype = datatype
         self.label = label
         self.css_id = css_id or self.name
         self.choices = choices or []
+        self.choice_provider = StaticChoiceProvider(self.choices)
```

Comment:
ah - yeah it's not very clear. basically by changing [this line](https://github.com/dimagi/commcare-hq/pull/14276/files#diff-0444f7ab4682ee00249d717a22b70783L341) `ChoiceListFilter` now adhere's to the spec needed for deferred_filters, which includes having a `choice_provider` that gets called elsewhere that didn't change, e.g. [here](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/app_manager/fixtures/mobile_ucr.py#L117). A lot of duck typing happening.